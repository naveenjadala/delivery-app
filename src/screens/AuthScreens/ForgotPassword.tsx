import {
  KeyboardAvoidingView,
  Platform,
  SafeAreaView,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import React from 'react';
import HeaderTitle from '../../components/common/HeaderTitle';
import SubmitBtn from '../../components/Buttons/SubmitBtn';
import CustomeHeader from '../../components/Headers/CustomeHeader';
import { useNavigation } from '@react-navigation/native';
import { LoginScreenNavigationProps } from '../../navigation/NavTypes';
import { GrayShade } from '../../assets/color';
import CustomeFormikTextInput from '../../components/TextInputs/CustomeFormikTextInput';
import RegisterTxt from '../../components/common/RegisterTxt';

const ForgotPassword = () => {
  const navigation = useNavigation<LoginScreenNavigationProps>();
  return (
    <SafeAreaView style={{ flex: 1, backgroundColor: 'white' }}>
      <ScrollView
        contentContainerStyle={{ flexGrow: 1, margin: 30 }}
        keyboardShouldPersistTaps="handled">
        <CustomeHeader
          iconName={'chevron-left'}
          onPress={() => navigation.goBack()}
        />
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
          style={styles.container}>
          <View>
            <HeaderTitle title="Forgot Password?" />
            <Text>
              Don't worry! It occurs. Please enter the email address linked with
              your account.
            </Text>
            <View style={{ marginVertical: 20 }}>
              <CustomeFormikTextInput placeholder="Enter your email" />
            </View>

            <SubmitBtn lable="Send Code" />
          </View>
        </KeyboardAvoidingView>
      </ScrollView>
      <View>
        <RegisterTxt
          content={'Remember Password? '}
          redirectionTxt={'Login'}
          onPress={() => navigation.navigate('Login')}
        />
      </View>
    </SafeAreaView>
  );
};

export default ForgotPassword;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    //   alignItems: 'center',
  },
  formContainer: {
    marginTop: 20,
    width: '100%',
  },
  forgotPassSty: {
    color: GrayShade,
    fontSize: 14,
    fontFamily: 'Urbanist',
    fontWeight: '600',
    alignSelf: 'flex-end',
  },
});
