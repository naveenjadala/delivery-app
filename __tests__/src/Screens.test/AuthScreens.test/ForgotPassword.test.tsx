import React from 'react';
import { render, screen } from '@testing-library/react-native';
import ForgotPassword from '../../../../src/screens/AuthScreens/ForgotPassword';

jest.mock('@react-navigation/native', () => ({
  useNavigation: () => {
    return jest.fn;
  },
}));

test('take a snapshort of the screen', () => {
  render(<ForgotPassword />);
  expect(screen).toMatchSnapshot();
});

test('check the title', () => {
  render(<ForgotPassword />);
  const getLinkElement = screen.getByText('Forgot Password?');
  expect(getLinkElement.props.children).toBe('Forgot Password?');
});
