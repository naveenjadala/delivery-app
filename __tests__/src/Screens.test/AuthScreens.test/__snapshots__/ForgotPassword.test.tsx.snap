// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`take a snapshort of the screen 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "flexGrow": 1,
        "margin": 30,
      }
    }
    keyboardShouldPersistTaps="handled"
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 60,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "borderColor": "#000",
              "borderRadius": 10,
              "borderWidth": 1,
              "height": 40,
              "opacity": 1,
              "padding": 10,
              "width": 40,
            }
          }
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#000",
                  "fontSize": 20,
                },
                undefined,
                {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ÔÅì
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          [
            {
              "flex": 1,
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <View>
          <Text
            style={
              {
                "color": "#000",
                "fontSize": 54,
                "marginVertical": 10,
                "textAlign": "left",
              }
            }
          >
            Forgot Password?
          </Text>
          <Text
            style={
              {
                "color": "#8391A1",
                "fontSize": 28,
                "marginVertical": 10,
                "textAlign": "left",
              }
            }
          >
            Don't worry! It occurs. Please enter the email address linked with your account.
          </Text>
          <View
            style={
              {
                "marginVertical": 20,
              }
            }
          >
            <TextInput
              placeholder="Enter your email"
              style={
                {
                  "backgroundColor": "#F7F8F9",
                  "borderColor": "#000",
                  "borderRadius": 8,
                  "borderWidth": 0.5,
                  "height": 55,
                  "marginVertical": 15,
                  "padding": 10,
                }
              }
            />
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "black",
                "borderRadius": 10,
                "height": 55,
                "justifyContent": "center",
                "marginVertical": 10,
                "opacity": 1,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 33,
                }
              }
            >
              Send Code
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "margin": 20,
          "opacity": 1,
        }
      }
    >
      <Text>
        Remember Password? 
         
        <Text
          style={
            {
              "color": "#35C2C1",
              "fontWeight": "500",
              "textDecorationLine": "underline",
            }
          }
        >
          Login
        </Text>
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
